admin/server/projects.go

```diff
@@ -1859,6 +1883,53 @@ func (s *Server) hasAssetUsagePermission(ctx context.Context, id, orgID, ownerID
 	return asset.OrganizationID != nil && *asset.OrganizationID == orgID && asset.OwnerID == ownerID
 }
 
+// githubRepoIDForProject returns the github repo id for a project stored in the database.
+// For older projects this may be nil since the github repo id was not stored.
+// This function fetches the ID from github API and stores it in the database.
+// It assumes that the project is connected to github and necessary checks have been done by the caller.
+func (s *Server) githubRepoIDForProject(ctx context.Context, p *database.Project) (int64, error) {
+	if p.GithubRepoID != nil {
+		return *p.GithubRepoID, nil
+	}
+
+	client, err := s.admin.Github.InstallationClient(*p.GithubInstallationID)
+	if err != nil {
```

Comment:
Maybe just to be safe, can it return an error if `p.GithubInstallationID == nil` (sounds like it would be an internal error)?